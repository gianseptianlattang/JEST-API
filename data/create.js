const randomstring = require('randomstring');

export const request = () => ({
        name: 'morpheus',
        job: 'leader',
});

export const negativeCase = [
    {
        title: 'null on name field',
        request: {
          name: null,
        },
      },
      {
        title: 'empty string on name field',
        request: {
          name: '',
        },
      },
      {
        title: 'number on name field',
        request: {
          name: randomstring.generate({ charset: 'numeric', length: '10' }),
        },
      },
      {
        title: 'less than 3 char on name field',
        request: {
          name: randomstring.generate({ charset: 'alphabetic', length: '2' }),
        },
      },
      {
        title: 'greater than 30 char on name field',
        request: {
          name: randomstring.generate({ charset: 'alphabetic', length: '31' }),
        },
      },
      {
        title: 'symbol @ on name field',
        request: {
          name: `@${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol . on name field',
        request: {
          name: `.${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol , on name field',
        request: {
          name: `,${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol _ on name field',
        request: {
          name: `_${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ! on name field',
        request: {
          name: `!${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol / on name field',
        request: {
          name: `/${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ? on name field',
        request: {
          name: `?${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol # on name field',
        request: {
          name: `#${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol $ on name field',
        request: {
          name: `$${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
    
      {
        title: 'symbol % on name field',
        request: {
          name: `%${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ^ on name field',
        request: {
          name: `^${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol & on name field',
        request: {
          name: `&${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol * on name field',
        request: {
          name: `*${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'null on job field',
        request: {
          job: null,
        },
      },
      {
        title: 'empty string on job field',
        request: {
          job: '',
        },
      },
      {
        title: 'number on job field',
        request: {
          name: randomstring.generate({ charset: 'numeric', length: '10' }),
        },
      },
      {
        title: 'less than 3 char on job field',
        request: {
          job: randomstring.generate({ charset: 'alphabetic', length: '2' }),
        },
      },
      {
        title: 'greater than 30 char on job field',
        request: {
          job: randomstring.generate({ charset: 'alphabetic', length: '31' }),
        },
      },
      {
        title: 'symbol @ on job field',
        request: {
          job: `@${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol . on job field',
        request: {
          job: `.${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol , on job field',
        request: {
          job: `,${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol _ on job field',
        request: {
          job: `_${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ! on job field',
        request: {
          job: `!${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol / on job field',
        request: {
          job: `/${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ? on job field',
        request: {
          job: `?${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol # on job field',
        request: {
          job: `#${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol $ on job field',
        request: {
          job: `$${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
    
      {
        title: 'symbol % on job field',
        request: {
          job: `%${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ^ on job field',
        request: {
          job: `^${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol & on job field',
        request: {
          job: `&${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol * on job field',
        request: {
          job: `*${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
];

export const snapEx = {
    id: expect.any(String),
    createdAt: expect.any(String),
};