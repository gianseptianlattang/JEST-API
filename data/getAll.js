export const queryparam = () => ({
    page : 2
});

export const casequeryparam = [
    {
        title: 'page 1',
        request: {
          page: 1,
        },
    },
    {
        title: 'page 2',
        request: {
          page: 2,
        },
    },
];

//Untuk get biasanya cek negatif case token, namun reqres.in tidak pake token.
// export const negativeTokenRequest = [
//     {
//       desc: 'without token',
//       request: {
//         token: '',
//       },
//     },
//     {
//       desc: 'with null token',
//       request: {
//         token: null,
//       },
//     },
//     {
//       desc: 'with wrong token - Man1',
//       request: {
//         token: Man1,
//       },
//     },
//     {
//       desc: 'with wrong token - Admin1',
//       request: {
//         token: Admin1,
//       },
//     },
//     {
//       desc: 'with random token',
//       request: {
//         token: randomToken,
//       },
//     },
//   ];