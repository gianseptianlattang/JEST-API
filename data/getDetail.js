export const queryparam = () => ({
    id : 2
});

export const casequeryparamid = [
    {
        title: 'id 1',
        request: {
          id: 1,
        },
    },
    {
        title: 'id 2',
        request: {
          id: 2,
        },
    },
];

//Untuk get biasanya cek negatif case token, namun reqres.in tidak pake token.
// export const negativeTokenRequest = [
//     {
//       desc: 'without token',
//       request: {
//         token: '',
//       },
//     },
//     {
//       desc: 'with null token',
//       request: {
//         token: null,
//       },
//     },
//     {
//       desc: 'with wrong token - Man1',
//       request: {
//         token: Man1,
//       },
//     },
//     {
//       desc: 'with wrong token - Admin1',
//       request: {
//         token: Admin1,
//       },
//     },
//     {
//       desc: 'with random token',
//       request: {
//         token: randomToken,
//       },
//     },
//   ];