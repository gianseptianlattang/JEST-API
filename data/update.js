const randomstring = require('randomstring');

export const request = () => ({
        name: 'morpheus2',
        job: 'member',
});

//Untuk Update biasanya cek negatif case token, namun reqres.in tidak pake token.
// export const negativeTokenRequest = [
//     {
//       desc: 'without token',
//       request: {
//         token: '',
//       },
//     },
//     {
//       desc: 'with null token',
//       request: {
//         token: null,
//       },
//     },
//     {
//       desc: 'with wrong token - Man1',
//       request: {
//         token: Man1,
//       },
//     },
//     {
//       desc: 'with wrong token - Admin1',
//       request: {
//         token: Admin1,
//       },
//     },
//     {
//       desc: 'with random token',
//       request: {
//         token: randomToken,
//       },
//     },
//   ];

export const negativeCase = [
    {
        title: 'null on name field',
        request: {
          name: null,
        },
      },
      {
        title: 'empty string on name field',
        request: {
          name: '',
        },
      },
      {
        title: 'number on name field',
        request: {
          name: randomstring.generate({ charset: 'numeric', length: '10' }),
        },
      },
      {
        title: 'less than 3 char on name field',
        request: {
          name: randomstring.generate({ charset: 'alphabetic', length: '2' }),
        },
      },
      {
        title: 'greater than 30 char on name field',
        request: {
          name: randomstring.generate({ charset: 'alphabetic', length: '31' }),
        },
      },
      {
        title: 'symbol @ on name field',
        request: {
          name: `@${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol . on name field',
        request: {
          name: `.${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol , on name field',
        request: {
          name: `,${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol _ on name field',
        request: {
          name: `_${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ! on name field',
        request: {
          name: `!${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol / on name field',
        request: {
          name: `/${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ? on name field',
        request: {
          name: `?${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol # on name field',
        request: {
          name: `#${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol $ on name field',
        request: {
          name: `$${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
    
      {
        title: 'symbol % on name field',
        request: {
          name: `%${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ^ on name field',
        request: {
          name: `^${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol & on name field',
        request: {
          name: `&${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol * on name field',
        request: {
          name: `*${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'null on job field',
        request: {
          job: null,
        },
      },
      {
        title: 'empty string on job field',
        request: {
          job: '',
        },
      },
      {
        title: 'number on job field',
        request: {
          name: randomstring.generate({ charset: 'numeric', length: '10' }),
        },
      },
      {
        title: 'less than 3 char on job field',
        request: {
          job: randomstring.generate({ charset: 'alphabetic', length: '2' }),
        },
      },
      {
        title: 'greater than 30 char on job field',
        request: {
          job: randomstring.generate({ charset: 'alphabetic', length: '31' }),
        },
      },
      {
        title: 'symbol @ on job field',
        request: {
          job: `@${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol . on job field',
        request: {
          job: `.${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol , on job field',
        request: {
          job: `,${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol _ on job field',
        request: {
          job: `_${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ! on job field',
        request: {
          job: `!${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol / on job field',
        request: {
          job: `/${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ? on job field',
        request: {
          job: `?${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol # on job field',
        request: {
          job: `#${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol $ on job field',
        request: {
          job: `$${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
    
      {
        title: 'symbol % on job field',
        request: {
          job: `%${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol ^ on job field',
        request: {
          job: `^${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol & on job field',
        request: {
          job: `&${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
      {
        title: 'symbol * on job field',
        request: {
          job: `*${randomstring.generate({ charset: 'alphabetic', length: '4' })}`,
        },
      },
];

export const snapEx = {
    updatedAt: expect.any(String),
};