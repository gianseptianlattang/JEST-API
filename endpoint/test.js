const supertest = require('supertest');
require('dotenv').config();

const api = supertest('https://reqres.in');

export const create = (body) => api.post('/api/users')
  .set('Content-Type', 'application/json')
  .set('Accept', 'application/json')
  .send(body);

export const update = (body, id) => api.put(`/api/users/${id}`)
  .set('Content-Type', 'application/json')
  .set('Accept', 'application/json')
  .send(body);

  export const getAll = (queryparam) => api.get('/api/users?')
  .set('Content-Type', 'application/json')
  .set('Accept', 'application/json')
  .query(queryparam);

  export const getDetail = (id) => api.get(`/api/users/${id}`)
  .set('Content-Type', 'application/json')
  .set('Accept', 'application/json')

  export const deleteUser = (id) => api.delete(`/api/users/${id}`)
  .set('Content-Type', 'application/json')
  .set('Accept', 'application/json')