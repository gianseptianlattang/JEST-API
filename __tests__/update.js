import * as data from '../data/update';
import * as endpoint from '../endpoint/test';
// import * as schema from '../schema/create'; untuk jest terbaru tidak ada fungsi tomatchschema

let body;
let response;

describe('Update User -- Positif Case', () => {
//   beforeAll(async () => {
//     await base.seedingData('');
//   });
    test(`Update User using valid request`, async () => {
        body = data.request();
        response = await endpoint.update(body, 2);
        expect(response.status).toEqual(200);
        expect(response.body).toMatchSnapshot(data.snapEx);
        // expect(response.body).toMatchSchema(schema.positif);
    });
});

describe('Update User -- Negatif Case', () => {
    //   beforeAll(async () => {
    //     await base.seedingData('');
    //   });
        (data.negativeCase).forEach(async (element) => {
            test(`Update User using ${element.title}`, async () => {
                body = data.request();
                const param = Object.keys(element.request);
                body[param]= element.request[param];
                response = await endpoint.update(body, 2);
                expect(response.status).toEqual(200); //harusnya 400 (200 mengikuti reqres.in)
                // expect(response.body).toMatchSnapshot(data.snapEx); //tidak bisa snapshot karena data berubah-ubah dan mengikuti reqres.in
            });
        });
    });