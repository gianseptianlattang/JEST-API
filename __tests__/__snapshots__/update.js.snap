// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create User -- Positif Case Create User using valid request 1`] = `
Object {
  "job": "member",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using empty string on job field 1`] = `
Object {
  "job": "",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using empty string on name field 1`] = `
Object {
  "job": "member",
  "name": "",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using greater than 30 char on job field 1`] = `
Object {
  "job": "rbIEPXyfWFXSRnqOcwIQWCqwSxfjKQC",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using greater than 30 char on name field 1`] = `
Object {
  "job": "member",
  "name": "lbfZddPXhEZaQUxXUNJeMYLcQcRDElk",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using less than 3 char on job field 1`] = `
Object {
  "job": "QS",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using less than 3 char on name field 1`] = `
Object {
  "job": "member",
  "name": "Om",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using null on job field 1`] = `
Object {
  "job": null,
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using null on name field 1`] = `
Object {
  "job": "member",
  "name": null,
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using number on job field 1`] = `
Object {
  "job": "member",
  "name": "1182028435",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using number on name field 1`] = `
Object {
  "job": "member",
  "name": "5636089670",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol ! on job field 1`] = `
Object {
  "job": "!uAZp",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol ! on name field 1`] = `
Object {
  "job": "member",
  "name": "!BweB",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol # on job field 1`] = `
Object {
  "job": "#HGXk",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol # on name field 1`] = `
Object {
  "job": "member",
  "name": "#rSrC",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol $ on job field 1`] = `
Object {
  "job": "$YIIg",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol $ on name field 1`] = `
Object {
  "job": "member",
  "name": "$oiGd",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol % on job field 1`] = `
Object {
  "job": "%BGmc",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol % on name field 1`] = `
Object {
  "job": "member",
  "name": "%BfKZ",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol & on job field 1`] = `
Object {
  "job": "&yrBP",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol & on name field 1`] = `
Object {
  "job": "member",
  "name": "&xeAU",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol * on job field 1`] = `
Object {
  "job": "*dTwB",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol * on name field 1`] = `
Object {
  "job": "member",
  "name": "*cviq",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol , on job field 1`] = `
Object {
  "job": ",UoUR",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol , on name field 1`] = `
Object {
  "job": "member",
  "name": ",UdIH",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol . on job field 1`] = `
Object {
  "job": ".RnzA",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol . on name field 1`] = `
Object {
  "job": "member",
  "name": ".bGjH",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol / on job field 1`] = `
Object {
  "job": "/kBvf",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol / on name field 1`] = `
Object {
  "job": "member",
  "name": "/LHNz",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol ? on job field 1`] = `
Object {
  "job": "?tQpt",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol ? on name field 1`] = `
Object {
  "job": "member",
  "name": "?Jjer",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol @ on job field 1`] = `
Object {
  "job": "@Wvfu",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol @ on name field 1`] = `
Object {
  "job": "member",
  "name": "@qSpj",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol ^ on job field 1`] = `
Object {
  "job": "^bfiJ",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol ^ on name field 1`] = `
Object {
  "job": "member",
  "name": "^BuRb",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol _ on job field 1`] = `
Object {
  "job": "_HJad",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Negatif Case Update User using symbol _ on name field 1`] = `
Object {
  "job": "member",
  "name": "_idEC",
  "updatedAt": Any<String>,
}
`;

exports[`Update User -- Positif Case Update User using valid request 1`] = `
Object {
  "job": "member",
  "name": "morpheus2",
  "updatedAt": Any<String>,
}
`;
